docs/reference/files/pyproject-toml-file.rst

```diff
@@ -0,0 +1,62 @@
+.. _pyproject-toml-file:
+
+
+``pyproject.toml`` file
+=======================
+
+The ``pyproject.toml`` file in your charm's root directory is a typical
+Python ``pyproject.toml`` file.
+
+    See more: `pip |
+    pyproject.toml <https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/>`_
+
+This file is generated automatically by ``charmcraft init`` with the
+contents below:
```

Comment:
This could vary by which profile one uses. IDK if it's possible, but something that would be **amazing** would be if you were to include (probably under a dropdown or with tabs?) each actual Jinja template for the file based on the profile. E.g. for the [django-framework profile](https://github.com/canonical/charmcraft/blob/main/charmcraft/templates/init-django-framework/pyproject.toml.j2)

Reply:
We would do this (probably in a different PR) but seven elements is too much for a tab container. Is there any way we could logically group them into multiple containers?

Reply:
I'm going to close this for now since in practice it looks like these are the same, although this raises those questions for, say, `charmcraft.yaml` which does vary for every profile.